daq/runner.py

```diff
@@ -296,6 +317,22 @@ def initialize(self):
 
         LOGGER.debug('Done with initialization')
 
+    def _create_stream_monitor(self):
+        return stream_monitor.StreamMonitor(idle_handler=self._handle_system_idle,
+                                            loop_hook=self._loop_hook,
+                                            timeout_sec=20)  # Polling rate
+
+    def _create_native_gateway(self):
+        if not self._native_vlan:
```

Comment:
This function is odd; this will probably be even more convoluted to anyone not familiar with DAQ.  Is there another path that can be used to start a native gateway?

Reply:
Yes -- I agree that was particularly messy. I've refactored everything, so hopefully it's better now.